source/configuration_manual/mail_location/obox/dictmap.rst

```diff
@@ -44,6 +44,54 @@ Cassandra/sproxyd Example Configuration
    }
 
+Cassandra/azure Example Configuration
+---------------------------------------
+
+.. code-block:: none
+
+   mail_location = obox:%u:INDEX=~/:CONTROL=~/
+   plugin {
+     # Without lazy_expunge plugin:
+     obox_fs = fscache:2G:/var/cache/mails/%4Nu:dictmap:proxy:dict-async:cassandra ; azure:https://ACCOUNTNAME:SHARED_KEY_BASE64_URL_ESCAPED@CONTAINERNAME.blob.core.windows.net/?reason_header_max_length=200 ; refcounting-table:lockdir=/tmp:bucket-size=10000:bucket-cache=%h/buckets.cache:nlinks-limit=3:delete-timestamp=+10s:bucket-deleted-days=11:storage-objectid-prefix=%u/mail-storage/
+     # With lazy_expunge plugin:
+     #obox_fs = fscache:2G:/var/cache/mails/%4Nu:dictmap:proxy:dict-async:cassandra ; azure:https://ACCOUNTNAME:SHARED_KEY_BASE64_URL_ESCAPED@CONTAINERNAME.blob.core.windows.net/?reason_header_max_length=200 ; refcounting-table:bucket-size=10000:bucket-cache=%h/buckets.cache:nlinks-limit=3:delete-timestamp=+10s:bucket-deleted-days=11:storage-objectid-prefix=%u/mail-storage/
+
+     obox_index_fs = compress:zstd:3:dictmap:proxy:dict-async:cassandra ; azure:https://ACCOUNTNAME:SHARED_KEY_BASE64_URL_ESCAPED@CONTAINERNAME.blob.core.windows.net/?reason_header_max_length=200 ; diff-table:storage-passthrough-paths=full
+     fts_dovecot_fs = fts-cache:fscache:2G:/var/cache/fts/%4Nu:compress:zstd:3:dictmap:proxy:dict-async:cassandra ; azure:https://ACCOUNTNAME:SHARED_KEY_BASE64_URL_ESCAPED@CONTAINERNAME.blob.core.windows.net/?reason_header_max_length=200 ; dict-prefix=%u/fts/:storage-passthrough-paths=full
+   }
+
+Cassandra/aws-s3 Example Configuration
+---------------------------------------
+
+.. code-block:: none
+
+   mail_location = obox:%u:INDEX=~/:CONTROL=~/
+   plugin {
+     # Without lazy_expunge plugin:
+     obox_fs = fscache:2G:/var/cache/mails/%4Nu:dictmap:proxy:dict-async:cassandra ; aws-s3:https://bucket-name.s3.region.amazonaws.com/?region=region&auth_role=s3access&reason_header_max_length=200 ; refcounting-table:lockdir=/tmp:bucket-size=10000:bucket-cache=%h/buckets.cache:nlinks-limit=3:delete-timestamp=+10s:bucket-deleted-days=11:storage-objectid-prefix=%u/mail-storage/
+     # With lazy_expunge plugin:
+     #obox_fs = fscache:2G:/var/cache/mails/%4Nu:dictmap:proxy:dict-async:cassandra ; aws-s3:https://bucket-name.s3.region.amazonaws.com/?region=region&auth_role=s3access&reason_header_max_length=200 ; refcounting-table:bucket-size=10000:bucket-cache=%h/buckets.cache:nlinks-limit=3:delete-timestamp=+10s:bucket-deleted-days=11:storage-objectid-prefix=%u/mail-storage/
+
+     obox_index_fs = compress:zstd:3:dictmap:proxy:dict-async:cassandra ; aws-s3:https://bucket-name.s3.region.amazonaws.com/?region=region&auth_role=s3access&reason_header_max_length=200 ; diff-table:storage-passthrough-paths=full
```

Comment:
I don't like the inconsistencies between the individual example URLs,

- sproxyd uses `https://sproxyd.scality.example.com`
- azure uses `https://ACCOUNTNAME:SHARED_KEY_BASE64_URL_ESCAPED@CONTAINERNAME.blob.core.windows.net`
- aws-s3 uses `https://bucket-name.s3.region.amazonaws.com`
- s3 uses `https://ACCESSKEY:SECRET@s3.example.com`

I would prefer to read it, if it was of the same form, e.g.: `https://<key>:<key>:...@<storage-url>` or `https:://KEY1:KEY2:...@STORAGE_URL` or similar.

This would make parsing the individual examples easier, especially because the the `obox-fs` settings are unfortunately long.

Reply:
I unified them to have the UPPERCASED words needing to be replaced by the user. They can't be totally unified because some require secrets some not. Some require a fully custom URL some not.

How do you like it now?